### ISSUE TITLE
ParamSpec args/kwargs not bound when Generic class is provided as a function argument

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#1963]
**Describe the bug**
Pyright will not bound paramspec args/kwargs to the generic of a class provided in the same function.

**To Reproduce**
Given the code snippet seen below:

Pyright will return the following.

`test.py:32:13 - info: Type of "a" is "Foo[(a: int), str]"`
`test.py:35:9 - error: Argument of type "Foo[(a: int), str]" cannot be assigned to parameter "foo" of type "Foo[P@foo, T@foo]" in function "foo"
    TypeVar "P@Foo" is invariant
      Type "(a: int)" is incompatible with ParamSpec "P" (reportGeneralTypeIssues)`
`test.py:35:12 - error: Param spec "P@foo" has no bound value (reportGeneralTypeIssues)`
`test.py:37:13 - info: Type of "c" is "T@foo"`

**Expected behavior**
No errors would occur as `ParamSpec` `(a: int)` should be assignable to the generic `P`

**Screenshots or Code**
```python
from __future__ import annotations


from typing import Callable, Generic, TYPE_CHECKING, TypeVar


if TYPE_CHECKING:
    from typing_extensions import ParamSpec

    P = ParamSpec("P")
else:
    P = TypeVar("P")  # hacky runtime fix

T = TypeVar("T")


class Foo(Generic[P, T]):
    def __init__(self, func: Callable[P, T]) -> None:
        ...


def foo(foo: Foo[P, T], *args: P.args, **kwargs: P.kwargs) -> T:
    ...


def func(a: int) -> str:
    ...


a = Foo(func)

reveal_type(a)


c = foo(a, 2)

reveal_type(c)

```
![image](https://user-images.githubusercontent.com/8677174/121341457-06a01880-c964-11eb-9575-de05cad0c955.png)

**VS Code extension or command-line**
Tested using both Pylance 2021.6.0 and Pyright CLI 1.1.147

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.148, which I just published. It will also be included in the next release of pylance.
